internal/dialect/dialectquery/postgres.go

```diff
@@ -50,7 +50,7 @@ func (p *Postgres) TableExists(tableName string) string {
 		q := `SELECT EXISTS ( SELECT 1 FROM pg_tables WHERE schemaname = '%s' AND tablename = '%s' )`
 		return fmt.Sprintf(q, schemaName, tableName)
 	}
-	q := `SELECT EXISTS ( SELECT 1 FROM pg_tables WHERE tablename = '%s' )`
+	q := `SELECT EXISTS ( SELECT 1 FROM pg_tables WHERE schemaname = current_schema() AND tablename = '%s' )`
```

Comment:
If we go down this route, then we should also handle the edge case where `current_schema ()` returns `NULL`.

> Returns the name of the schema that is first in the search path (or a null value if the search path is empty). This is the schema that will be used for any tables or other named objects that are created without specifying a target schema.

E.g., `SET search_path TO '';`

So the suggestion is to update the predicate to:

```diff
- AND schemaname = current_schema()
+ AND (current_schema() IS NULL OR schemaname = current_schema())
```